src/fsharp/FSharp.Compiler.Unittests/HashIfExpression.fs

```diff
@@ -0,0 +1,268 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.Unittests
+
+open System
+open System.Text
+
+open NUnit.Framework
+
+open Internal.Utilities.Text.Lexing
+open Microsoft.FSharp.Compiler
+open Microsoft.FSharp.Compiler.Lexer
+open Microsoft.FSharp.Compiler.Lexhelp
+open Microsoft.FSharp.Compiler.ErrorLogger
+open Microsoft.FSharp.Compiler.ErrorLogger
+open Microsoft.FSharp.Compiler.Ast
+
+[<TestFixture>]
+type HashIfExpression()     =
+
+    let preludes    = [|"#if "; "#elif "|]
```

Comment:
Although that won't happen in the compiler currently because lex.fsl guarantee correct prelude.